bin/fetch-configlet.ps1

```diff
@@ -1,26 +1,34 @@
 $ErrorActionPreference = "Stop"
 $ProgressPreference = "SilentlyContinue"
 
-$requestOpts = @{
-    Headers = If ($env:GITHUB_TOKEN) { @{ Authorization = "Bearer ${env:GITHUB_TOKEN}" } } Else { @{ } }
-    MaximumRetryCount = 3
-    RetryIntervalSec = 1
+if ($PSVersionTable.PSVersion.Major -le 5) {
+    # PreserveAuthorizationOnRedirect requires PowerShell Core 7+
+    # TODO create a tool that install pwsh in the directory https://docs.microsoft.com/en-us/dotnet/core/tools/local-tools-how-to-use
+    throw "Please run with PowerShell Core 'pwsh'. If you need to install pwsh globally: 'dotnet tool install --global PowerShell'"
 }
 
-$arch = If ([Environment]::Is64BitOperatingSystem) { "64bit" } Else { "32bit" }
-$fileName = "configlet-windows-$arch.zip"
-
-Function Get-DownloadUrl {
+function Get-DownloadUrl($FileName, $RequestOpts) {
     $latestUrl = "https://api.github.com/repos/exercism/configlet/releases/latest"
-    Invoke-RestMethod -Uri $latestUrl -PreserveAuthorizationOnRedirect @requestOpts
-    | Select-Object -ExpandProperty assets
-    | Where-Object { $_.browser_download_url -match $FileName }
-    | Select-Object -ExpandProperty browser_download_url
+    $assets = Invoke-RestMethod -Uri $latestUrl -PreserveAuthorizationOnRedirect @RequestOpts | Select-Object -ExpandProperty assets
+    return $assets | Where-Object { $_.browser_download_url -match $FileName } | Select-Object -ExpandProperty browser_download_url -First 1
```

Comment:
Was the return a suggestion by the analyzer?

Reply:
Same question for the -First 1

Reply:
So the analyzer was unhappy with the pipe with no argument before.

```
ParseError [16 , 5] fetch-configlet.ps1 EmptyPipeElement - An empty pipe element is not allowed.
```

If you prefer to rollback or have another idea, I am good with both.